google/auth/aws.py

```diff
@@ -406,6 +406,7 @@ def __init__(
         self._cred_verification_url = credential_source.get(
             "regional_cred_verification_url"
         )
+        self._aws_token_url = credential_source.get("aws_token_url")
```

Comment:
Existing config files won't contain aws_token_url. So maybe adding a default value as backup can help?
Or maybe we can skip getting the session token if the aws_token_url is not provided and incase IDMSv2 is used, we can error out